apps/api/v2/src/modules/billing/billing.processor.ts

```diff
@@ -0,0 +1,76 @@
+import { BillingRepository } from "@/modules/billing/billing.repository";
+import { OrganizationsRepository } from "@/modules/organizations/organizations.repository";
+import { StripeService } from "@/modules/stripe/stripe.service";
+import { Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+
+export const INCREMENT_JOB = "increment";
+export const BILLING_QUEUE = "billing";
+export type IncrementJobDataType = {
+  userId: number;
+  bookingId?: number | null;
+  startTime?: Date | null;
+  endTime?: Date | null;
+};
+
+export type DecrementJobDataType = IncrementJobDataType;
+
+@Processor(BILLING_QUEUE)
+export class BillingProcessor {
+  private readonly logger = new Logger(BillingProcessor.name);
+
+  constructor(
+    public readonly stripeService: StripeService,
+    private readonly billingRepository: BillingRepository,
+    private readonly teamsRepository: OrganizationsRepository
+  ) {}
+
+  @Process(INCREMENT_JOB)
+  async handleIncrement(job: Job<IncrementJobDataType>) {
+    const { userId } = job.data;
+    try {
+      // get the platform org of the managed user
+      const team = await this.teamsRepository.findPlatformOrgFromUserId(userId);
+      const teamId = team.id;
+      if (!team.id) return void 0;
```

Comment:
Each time we `return void 0;` in `handleIncrement` - should we throw an error with a corresponding message? Because if the job is called to increase billing but something is not met then we should catch it. Or the `handleIncrement` should not be called if any of the cases with `return void 0` happens.

Reply:
we don't need to return an error since jobs are executed in the background but we should definitely log the issue for debugging in the future